### PR TITLE
documented prompt Text area

### PR DESCRIPTION
**Pull request by khushishikhu**
closes odk-x/tool-suite-X#152

addresses odk-x/tool-suite-X#152

#### What is included in this PR?
It includes the prompt for the text area if we go above 255 characters. I have added the way to increase the character limit by adding the note there

#### What is left to be done in the addressed issue?
Please tell me if I left anything or what more it requires.
Thank you!!

### CODE REVIEW

File: odkx-src/xlsx-converter-reference.rst [hunk 1]
```diff
@@ -388,6 +388,21 @@ The following prompt types are available in ODK-X Survey.
       | like to enter.
   * - video
     - | Used to capture a video.
+  * - Text Area
+    - | Used to allow the user to write upto 255 characters set by default
```
Comment (wbrunette) [inline odkx-src/xlsx-converter-reference.rst:392]: This is incorrect, the text area provides a big text area for users to enter information. The note should be if you anticipate users writing anything longer than 255 characters you need to change the model sheet and change the elementType column. As the documentation states: "For example, as shown in the datatypes XLSX, string variables' length can be adjusted from a default of 255 to other lengths with string(len)." https://docs.odk-x.org/xlsx-converter-reference/#model

Comment (wbrunette) [reply odkx-src/xlsx-converter-reference.rst:392]: This statement was confusing ...

**Review by wbrunette — CHANGES_REQUESTED** [review]

File: odkx-src/xlsx-converter-reference.rst [hunk 2]
```diff
@@ -392,7 +392,7 @@ The following prompt types are available in ODK-X Survey.
     - | Used to enter the information in a big text area or pragraphs.
     
 .. note::
-   if users anticipates for writing anything longer than 255 characters then the user need to change the model sheet and change the elementType column. As the documentation states: "For example, as shown in the datatypes XLSX, string variables' length can be adjusted from a default of 255 to other lengths with string(len)". For example, if you had a string prompt named long_data that you wanted to be 500 characters, you would add the following to your model worksheet.
+   if users anticipates for writing anything longer than 255 characters then the user need to change the model sheet and change the element-type column. As the documentation states: "For example, as shown in the datatypes XLSX, string variables' length can be adjusted from a default of 255 to other lengths with string(len)". For example, if you had a string prompt named long_data that you wanted to be 500 characters, you would add the following to your model worksheet.
```
Comment (elmps2018) [inline odkx-src/xlsx-converter-reference.rst:395]: At least in the current documentation, the column is named elementType, not "element-type" so I think this change may be incorrect...

File: odkx-src/xlsx-converter-reference.rst [hunk 3]
```diff
@@ -393,7 +393,7 @@ The following prompt types are available in ODK-X Survey.
     
 .. note::
   
-  To increase the length for a string, you could add a model sheet to your xlsx workbook and set the elementType to be string(<string_length>).For example, if you had a string
+  To increase the length for a string, you could add a model sheet to your xls workbook and set the element-type to be string(<string_length>).For example, if you had a string
```
[the remark this answers is not in the document]
Comment (khushishikhu) [reply odkx-src/xlsx-converter-reference.rst]: @elmps2018 I have written xlsx workbook only but then it is not passing the build

File: odkx-src/xlsx-converter-reference.rst (hunk 2 above)
Comment (khushishikhu) [reply odkx-src/xlsx-converter-reference.rst:395]: yes, I have written "elementType" only but the build is not passing. That shows it as spelling error

File: odkx-src/xlsx-converter-reference.rst [hunk 4]
```diff
@@ -388,6 +388,21 @@ The following prompt types are available in ODK-X Survey.
       | like to enter.
   * - video
     - | Used to capture a video.
+  * - Text Area
+    - | Used to allow the user to write upto 255 characters set by default
+    
+.. note::
+  
+  To increase the length for a string, you could add a model sheet to your xls workbook and set the element-type to be string(<string_length>).For example, if you had a string
```
[the remark this answers is not in the document]
Comment (khushishikhu) [reply odkx-src/xlsx-converter-reference.rst]: Yes, The text area is a string prompt, as we can not write after covering 255 characters.

File: odkx-src/xlsx-converter-reference.rst [hunk 5]
```diff
@@ -388,6 +388,19 @@ The following prompt types are available in ODK-X Survey.
       | like to enter.
   * - video
     - | Used to capture a video.
+  * - Text Area
+    - | Used to enter the information in a big text area or pragraphs.
+    
+.. note::
+   if users anticipates for writing anything longer than 255 characters then the user need to change the model sheet and change the elementType column. As the documentation states: "For example, as shown in the datatypes XLSX, string variables' length can be adjusted from a default of 255 to other lengths with string(len)". For example, if you had a string prompt named long_data that you wanted to be 500 characters, you would add the following to your model worksheet.
```
Comment (wbrunette) [inline odkx-src/xlsx-converter-reference.rst:395]: I like the rest of the changes! But I think this is weird: "As the documentation states: "For example, as shown in the datatypes XLSX, string variables' length can be adjusted from a default of 255 to other lengths with string(len)". as this IS the documentation so seems weird. That was my comment to you to help you update your pull request.

**Review by wbrunette — CHANGES_REQUESTED** [review]

File: odkx-src/xlsx-converter-reference.rst (hunk 2 above)
Comment (wbrunette) [reply odkx-src/xlsx-converter-reference.rst]: you can add the word to https://github.com/odk-x/docs/blob/master/odkx-src/spelling_wordlist.txt and the spell checker will recognize it (there is an issue already to document this that someone is working on https://github.com/odk-x/tool-suite-X/issues/247)

File: odkx-src/xlsx-converter-reference.rst [hunk 6]
```diff
@@ -388,6 +388,19 @@ The following prompt types are available in ODK-X Survey.
       | like to enter.
   * - video
     - | Used to capture a video.
+  * - Text Area
```
Comment (linl33) [inline odkx-src/xlsx-converter-reference.rst:391]: This should be the identifier for the prompt type, which is `textarea`.

File: odkx-src/xlsx-converter-reference.rst [hunk 7]
```diff
@@ -388,6 +388,19 @@ The following prompt types are available in ODK-X Survey.
...
+    - | Used to enter the information in a big text area or pragraphs.
```
Comment (linl33) [inline odkx-src/xlsx-converter-reference.rst:392]: ~~pragraphs~~
paragraphs

File: odkx-src/xlsx-converter-reference.rst [hunk 8]
```diff
@@ -388,6 +388,19 @@ The following prompt types are available in ODK-X Survey.
...
+    
+.. note::
+   if users anticipates for writing anything longer than 255 characters then the user needs to change the model sheet and change the elementType column. It is shown in the datatypes XLSX, string variables' length can be adjusted from a default of 255 to other lengths with string(len). For example, if you had a string prompt named long_data that you wanted to be 500 characters, you would add the following to your model worksheet.
```
Comment (linl33) [inline odkx-src/xlsx-converter-reference.rst:395]: It would be helpful to link to the **model** section.

**Review by linl33 — CHANGES_REQUESTED** [review]
Thanks for documenting this.

File: odkx-src/xlsx-converter-reference.rst (hunk 2 above)
Comment (khushishikhu) [reply odkx-src/xlsx-converter-reference.rst]: @wbrunette I have added the word and made the PR for that https://github.com/odk-x/docs/pull/82 please review that

**Comment by elmps2018** [comment]
@khushishikhu I reviewed and merged the spelling list changes.

**Comment by khushishikhu** [comment]
> @khushishikhu I reviewed and merged the spelling list changes.

@elmps2018  The words are being added, but still build isn't passing

**Comment by elmps2018** [comment]
@khushishikhu not sure why this is still not passing after the spelling merged... @wbrunette or @linl33 any ideas?

**Comment by wbrunette** [comment]
@khushishikhu you need to merge your changes to ODK-X master into your current branch to get the updated code in it.

@khushishikhu and @elmps2018 the simpler process is to add your spelling change into the pull request you are proposing so that it is automatically incorporated into the build. 

At this point you need to merge in from ODK-X master to avoid any conflicts later.

**Comment by wbrunette** [comment]
@khushishikhu remember you can run the build commands locally on your computer so you don't have to keep pushing small commits. Instead of just running odkx-autobuild there are more commands: https://github.com/odk-x/docs#build-tasks

**Comment by khushishikhu** [comment]
@wbrunette yes sure!! I will take care of this thing from next time. Now the build is passing.

File: odkx-src/xlsx-converter-reference.rst [hunk 9]
```diff
@@ -391,6 +391,19 @@ The following prompt types are available in ODK-X Survey.
...
+  * - textarea
+    - | Used to enter the information in a big text area or paragraphs.
+    
+.. note::
+   if users anticipates for writing anything longer than 255 characters then the user needs to change the model sheet and change the elementType column. It is shown in the datatypes XLSX, string variables' length can be adjusted from a default of 255 to other lengths with string(len). For example, if you had a string prompt named long_data that you wanted to be 500 characters, you would add the following to your model worksheet. To know more refer to https://docs.odk-x.org/xlsx-converter-reference/#model
+.. list-table:: 
+
+  * - | name      
+    - | type   
+    - | element type
```
Comment (linl33) [inline odkx-src/xlsx-converter-reference.rst:403]: This should be `elementType`

File: odkx-src/xlsx-converter-reference.rst [hunk 10]
```diff
@@ -391,6 +391,19 @@ The following prompt types are available in ODK-X Survey.
       | like to enter.
   * - video
     - | Used to capture a video.
+  * - textarea
+    - | Used to enter the information in a big text area or paragraphs.
+    
+.. note::
+   if users anticipates for writing anything longer than 255 characters then the user needs to change the model sheet and change the elementType column. It is shown in the datatypes XLSX, string variables' length can be adjusted from a default of 255 to other lengths with string(len). For example, if you had a string prompt named long_data that you wanted to be 500 characters, you would add the following to your model worksheet. To know more refer to https://docs.odk-x.org/xlsx-converter-reference/#model
```
Comment (linl33) [inline odkx-src/xlsx-converter-reference.rst:398]: Please use a `ref` to link to that section. https://www.sphinx-doc.org/en/master/usage/restructuredtext/roles.html#cross-referencing-arbitrary-locations

**Review by wbrunette — APPROVED** [review]

**Pull request merged**